extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -742,6 +729,27 @@ export class AmpAutocomplete extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Requests remote data source, if provided, on first user interaction.
+   * @return {!Promise}
+   * @private
+   */
+  checkFirstInteractionAndMaybeFetchData_() {
+    if (!this.interacted_ && this.element.hasAttribute('src')) {
```

Comment:
This is a bit cleaner using a [guard clause](https://en.wikipedia.org/wiki/Guard_(computer_science)). For example:

```javascript
if (this.interacted || !this.element.hasAttribute('src')) {
  return Promise.resolve();
}

...
```